streampipes-extensions/streampipes-connect-adapters-iiot/src/main/java/org/apache/streampipes/connect/iiot/protocol/stream/FileReplayAdapter.java

```diff
@@ -133,66 +155,113 @@ public void onAdapterStarted(IAdapterParameterExtractor extractor,
+            .toList();
+
+    return renamingRules.stream()
+        .filter(rule -> rule.getNewRuntimeKey()
+            .equals(timestampRuntimeName))
+        .toList();
+  }
+
+  private void parseFile(
+      IAdapterParameterExtractor extractor,
+      IEventCollector collector,
+      IAdapterRuntimeContext adapterRuntimeContext
+  ) {
     try {
       InputStream inputStream = getDataFromEndpoint(extractor
           .getStaticPropertyExtractor()
           .selectedFilename(FILE_PATH));
 
-      extractor.selectedParser().parse(inputStream, (event) -> {
-
-        long actualEventTimestamp = -1;
-        if (event.get(timestampRuntimeName) instanceof Long) {
-          actualEventTimestamp = (long) event.get(timestampRuntimeName);
-        } else {
-          LOG.error(
-              "The timestamp field is not a unix timestamp in ms. Value: %s"
-                  .formatted(event.get(timestampRuntimeName)));
-        }
-
-        if (timestampLastEvent != -1) {
-          long sleepTime = (long) ((actualEventTimestamp - timestampLastEvent) / speedUp);
-          // speed up is set to Float.MAX_VALUE when user selected fastest option
-          if (sleepTime > 0 && speedUp != Float.MAX_VALUE) {
-            try {
-              Thread.sleep(sleepTime);
-            } catch (InterruptedException e) {
-              LOG.info("File stream adapter was stopped, the current replay is interuppted", e);
+      extractor.selectedParser()
+          .parse(inputStream, (event) -> {
+
+            long actualEventTimestamp = -1;
+            if (event.get(timestampSourceFieldName) instanceof Long) {
+              actualEventTimestamp = (long) event.get(timestampSourceFieldName);
+            } else {
+              LOG.error(
+                  "The timestamp field is not a unix timestamp in ms. Value: %s"
+                      .formatted(event.get(timestampSourceFieldName)));
             }
-          }
-        }
 
-        timestampLastEvent = actualEventTimestamp;
-        if (replaceTimestamp) {
-          event.put(timestampRuntimeName, System.currentTimeMillis());
-        }
+            if (timestampLastEvent != -1) {
+              long sleepTime = (long) ((actualEventTimestamp - timestampLastEvent) / speedUp);
+              // speed up is set to Float.MAX_VALUE when user selected fastest option
+              if (sleepTime > 0 && speedUp != Float.MAX_VALUE) {
+                try {
+                  Thread.sleep(sleepTime);
+                } catch (InterruptedException e) {
+                  LOG.info("File stream adapter was stopped, the current replay is interuppted", e);
+                }
+              }
+            }
+
+            timestampLastEvent = actualEventTimestamp;
+            if (replaceTimestamp) {
+              // we can directly use the desired runtime name for the timestamp instead of the original here
+              event.put(timestampRuntimeName, System.currentTimeMillis());
```

Comment:
Have you checked how this affects the `Rename timestap` rule? Looking at the `RenameTransformationRule` rule, I'm not sure if the new field `timestamp` gets the value `null` when `replaceTimestamp` is active

Reply:
I'll have a look

Reply:
I think it won't be `null` but it will probably be overwritten with the old value.
I'll change it.